packages/engine/src/worker/mod.rs

```diff
@@ -193,12 +206,32 @@ impl WorkerController {
                 }
             }
         }
-        // py_handle.await??;
+
         js_handle.await??;
+        // py_handle.await??;
+        // rs_handle.await??;
+
         Ok(())
     }
 
-    /// TODO: DOC
+    /// Handles a message from the worker pool.
+    ///
+    /// Depending of the content of the message, the following actions are executed:
+    ///   - [`Task`]: The contained task is spawned for the specified simulation, [`TaskMsg`] is
```

Comment:
I appreciate the method is called `spawn_task` (not sure why), but I don't really know what
> The contained task is spawned for the specified simulation

even means.

It looks to me like the method wraps the task and forwards it to the appropriate language runner